components/workflow-mgt/org.wso2.carbon.identity.workflow.impl/src/main/java/org/wso2/carbon/identity/workflow/impl/WorkflowImplServiceImpl.java

```diff
@@ -189,46 +181,50 @@ public void deleteHumanTask(WorkflowRequest workflowRequest) throws WorkflowImpl
             input.setPageNumber(0);
             input.setSimpleQueryCategory(TSimpleQueryCategory.ALL_TASKS);
             for (int i = 0; i < bpsProfiles.size(); i++) {
+                try {
                 String host = bpsProfiles.get(i).getWorkerHostURL();
-                URL servicesUrl = new URL(new URL(host), WFImplConstant.HT_SERVICES_URL);
-                stub = new HumanTaskClientAPIAdminStub(servicesUrl.toString());
-                ServiceClient client = stub._getServiceClient();
+                String servicesUrl = host + WFImplConstant.HT_SERVICES_URL;
```

Comment:
Yes, we iterate this loop for each 'bps profile' and bps profiles have different urls.